### PR TITLE
Cleanup UI code after Vaadin migration

### PR DESCRIPTION
**Pull request by blomark**
Refactored code and removed duplications in WindowController classes and Layout classes

This PR also contains a bugfix when updating an existing distribution set a duplicate error is no longer shown

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Review by bogdan-bondar — CHANGES_REQUESTED** [review]
Couple of small comments, but otherwise great job! ;)

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/data/filters/DsFilterParams.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.io.Serializable;
+
+import org.eclipse.hawkbit.ui.common.data.providers.DistributionSetManagementStateDataProvider;
+import org.springframework.util.StringUtils;
+
+/**
+ * Filter params for {@link DistributionSetManagementStateDataProvider}.
+ */
+public class DsFilterParams implements Serializable {
```
[the remark this answers is not in the document]
Comment (blomark) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/data/filters/DsFilterParams.java:19]: I would suggest to discuss this independantly of this PR

**Review by bogdan-bondar — APPROVED** [review]
Thanks a lot @blomark!

**Pull request merged**